DatadogLogs/Sources/Logs.swift

```diff
@@ -0,0 +1,113 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+import DatadogInternal
+
+/// iOS Log Collection
+///
+/// Send logs to Datadog from your iOS applications with Datadog’s dd-sdk-ios client-side logging library and leverage the following features:
+/// - Log to Datadog in JSON format natively.
+/// - Use default and add custom attributes to each log sent.
+/// - Record real client IP addresses and User-Agents.
+/// - Leverage optimized network usage with automatic bulk posts.
+public struct Logs {
+    /// The Logs general configuration.
+    ///
+    /// This configuration will be applied to all Logger instances.
+    public struct Configuration {
+        public typealias EventMapperClosure = (LogEvent) -> LogEvent?
+        /// Sets the sampling rate for logging.
+        ///
+        /// The sampling rate must be a value between `0` and `100`. A value of `0` means no logs will be processed, `100`
+        /// means all logs will be processed.
+        ///
+        /// By default sampling is disabled, meaning that all logs are being processed).
+        public var samplingRate: Float
+
+        /// Overrides the custom server endpoint where Logs are sent.
+        public var customIntakeURL: URL?
+
+        /// Overrides the main bundle instance.
+        public var bundle: Bundle = .main
+
+        /// Overrides the current process info.
+        public var processInfo: ProcessInfo = .processInfo
+
+        /// Sets the custom mapper for `LogEvent`. This can be used to modify logs before they are send to Datadog.
+        ///
+        /// - Parameter mapper: the closure taking `LogEvent` as input and expecting `LogEvent` as output.
+        /// The implementation should obtain a mutable version of the `LogEvent`, modify it and return it. Returning `nil` will result
+        /// with dropping the Log event entirely, so it won't be send to Datadog.
+        public mutating func setEventMapper(_ mapper: @escaping EventMapperClosure) {
+            eventMapper = SyncLogEventMapper(mapper)
+        }
```

Comment:
convention/ Why not `public var eventMapper: EventMapperClosure? = nil`? If we need to wrap it into `SyncLogEventMapper`, this can happen when resolving the config in `Logs.enable(with:in:)` or even later. Exposing a setter looks very inconsistent and doesn't promote simplicity.

Reply:
Currently, we only allow mapper in sync so we expose the closure. The async mapper is used by Flutter, therefor the 'internal mutation' which need to be set on the config object.

But I agree that a property would look better. We can do:

```swift

public var eventMapper: EventMapperClosure? {
  get { (_eventMapper as? SyncLogEventMapper)?.mapper }
  set { _eventMapper = SyncLogEventMapper(newValue) }
}

internal var _eventMapper: LogEventMapper? = nil
```

Not sure which one I prefer 🤔